### PR TITLE
Speedy: group compiler parameters into a Config case class.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Speedy compiler is parametrized by several option flags.  Those flags
are passed through different calls (in particular in the instance of
`CompiledPackages`.

This PR group all this configuration parameters into a case class to
pass the option in a cleaner way.

This PR is based on #7436.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by nickchapman-da — APPROVED** [review]
LGTM . Thanks!

File: daml-lf/interpreter/perf/src/main/scala/com/daml/lf/explore/Explore.scala [hunk 1]
```diff
@@ -62,7 +65,7 @@ object PlaySpeedy {
   }
 
   private val noPackages =
-    data.assertRight(PureCompiledPackages(Map.empty, Compiler.NoStackTrace, Compiler.NoProfile))
+    PureCompiledPackages(Map.empty, Map.empty, compilerConfig)
```
Comment (cocreature) [inline daml-lf/interpreter/perf/src/main/scala/com/daml/lf/explore/Explore.scala:68]: afaict, we had disabled stack tracing here before now it’s enabled. I guess it doesn’t make a difference?

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SpeedyTest.scala [hunk 2]
```diff
@@ -316,6 +318,14 @@ class SpeedyTest extends WordSpec with Matchers {
 
     }
 
+    /*
+    Some(SELabelClosure(LfDefRef(-pkgId-:M:p_1_0),
+        SELet1General(SEVal(LfDefRef(-pkgId-:M:origin)),SEAppAtomicSaturatedBuiltin(SBRecUpd(-pkgId-:M:Point,0),[SELocS(1),SEValue(SInt64(1))])))) did not equal
+    Some(SELet1General(SEVal(LfDefRef(-pkgId-:M:origin)),SEAppAtomicSaturatedBuiltin(SBRecUpd(-pkgId-:M:Point,0),[SELocS(1),SEValue(SInt64(1))])))
```
Comment (hurryabit) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SpeedyTest.scala:324]: Why is this test commented out?

**Review by hurryabit — COMMENTED** [review]
Excellent idea. Thank you.

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SpeedyTest.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SpeedyTest.scala:324]: They have nothing to do here. 
I remove them in #7450

Comment (hurryabit) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SpeedyTest.scala:324]: I see. Thanks for the explanation.

**Pull request merged**